### PR TITLE
[INCR-79] upgrade django-oauth2-provider

### PR DESCRIPTION
**Pull request by singuliere**

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @singuliere! I've created [OSPR-3086](https://openedx.atlassian.net/browse/OSPR-3086) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

File: requirements/dev.txt [hunk 1]
```diff
@@ -4,83 +4,70 @@
...
 #    pip-compile --output-file requirements/dev.txt requirements/base.in requirements/dev.in requirements/quality.in requirements/test.in requirements/travis.in
 #
--e git+https://github.com/edx/django-oauth2-provider.git@0.2.7-fork-edx-6a#egg=django-oauth2-provider==0.2.7-fork-edx-6
 -e git+https://github.com/edx/django-rest-framework-oauth.git@f0b503fda8c254a38f97fef802ded4f5fe367f7a#egg=djangorestframework-oauth
 -e git+https://github.com/edx/django-rest-framework.git@1ceda7c086fddffd1c440cc86856441bbf0bd9cb#egg=djangorestframework==3.6.3
 appdirs==1.4.3            # via fs
 argparse==1.4.0           # via caniusepython3
-asn1crypto==0.24.0        # via cryptography
 astroid==1.5.3            # via edx-lint, pylint, pylint-celery
-backports.functools-lru-cache==1.5  # via astroid, caniusepython3, pylint
```
Comment (jmbowman) [inline requirements/dev.txt]: I'm guessing you ran `make upgrade` under Python 3?  Given that none of the packages depending on this were upgraded, I think this is still needed under Python 2.7.  We'll probably need to keep running `make upgrade` under 2.7 until either we drop support for it or `pip-tools` gets smarter about noticing different requirements under different Python versions.  I suspect several of the other dependencies that were removed from these files are in a similar situation.

Comment (singuliere) [reply requirements/dev.txt]: You guessed right and I did not realize that makes a difference. Thanks a lot for catching this!

**Comment by natabene** [comment]
@singuliere Thank you for your contribution.  @edx/incr-reviewers Can you give this a look?

**Comment by edx-webhook** [comment]
@singuliere 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/edx-val&entry.867055334=https://github.com/edx/edx-val/pull/169&entry.1484655318=https://github.com/singuliere&entry.752974735=2019-02-17+16:19&entry.1917517419=2019-02-19+20:28&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**